sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/ClassAndMethodName.cs

```diff
@@ -85,6 +87,14 @@ protected override void Initialize(SonarAnalysisContext context)
                 },
                 SyntaxKind.MethodDeclaration);
 
+            context.RegisterSyntaxNodeActionInNonGenerated(
+                c =>
+                {
+                    var declaration = (LocalFunctionStatementSyntaxWrapper)c.Node;
+                    CheckMemberName(declaration, declaration.Identifier, c);
+                },
+                SyntaxKindEx.LocalFunctionStatement);
```

Comment:
In order to remove the code duplication we could merge the 3 calls of RegisterSyntaxNodeActionInNonGenerated and extract a function which will return the identifier based on the node type.